Add DISCO switch staging configs

stephen-soltesz opened the pull request:
This change adds a staging-specific app.yaml config for running the disco parser in staging.

This change also adds explicit environment variables to the sandbox config.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/etl/432)
<!-- Reviewable:end -->

gfr10598 commented:
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[.travis.yml, line 114 at r1](https://reviewable.io:443/reviews/m-lab/etl/432#-L4mBkjm37DTS6_NPv4n:-L4mBkjm37DTS6_NPv4o:bn4r7uk) ([raw file](https://github.com/m-lab/etl/blob/8713afb4a6f05ba27e6178386f76ed6dc0df17e3/.travis.yml#L114)):*
> ```YAML
>     /tmp/mlab-staging.json $TRAVIS_BUILD_DIR/cmd/etl_worker app-sidestream-staging.yaml
>     && $TRAVIS_BUILD_DIR/travis/deploy_app.sh mlab-staging
>     /tmp/mlab-staging.json $TRAVIS_BUILD_DIR/cmd/etl_worker app-disco-staging.yaml
> ```

What do you think about injecting things that differ between sandbox/staging/prod, instead of having separate yaml files?
I started this with gardener, and would like to work out whether to standardize on it.

---

*[cmd/etl_worker/app-disco-staging.yaml, line 42 at r1](https://reviewable.io:443/reviews/m-lab/etl/432#-L4mBy9TGMgkFD-1-RmG:-L4mBy9UHr3akYuns19B:b-7prfl8) ([raw file](https://github.com/m-lab/etl/blob/8713afb4a6f05ba27e6178386f76ed6dc0df17e3/cmd/etl_worker/app-disco-staging.yaml#L42)):*
> ```YAML
>   MAX_WORKERS: 20
>   BIGQUERY_PROJECT: 'mlab-staging'
>   BIGQUERY_DATASET: 'staging'
> ```

I think we should actually use the same dataset in every project.  Different for batch vs daily, but otherwise the same across projects.  so - base_tables - Make sense?

---

*[cmd/etl_worker/app-disco.yaml, line 42 at r1](https://reviewable.io:443/reviews/m-lab/etl/432#-L4mCJ7ONWANfgOJVLCx:-L4mCJ7ONWANfgOJVLCy:bl04edp) ([raw file](https://github.com/m-lab/etl/blob/8713afb4a6f05ba27e6178386f76ed6dc0df17e3/cmd/etl_worker/app-disco.yaml#L42)):*
> ```YAML
>   MAX_WORKERS: 20
>   BIGQUERY_PROJECT: 'mlab-sandbox'
>   BIGQUERY_DATASET: 'mlab_sandbox'
> ```

base_tables?  Or incoming?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/432)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>
Some optional suggestions, that would help start converging.

---

Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/432#-:-L4mCPoeZnGs41gJpBob:b-we596h)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[.travis.yml, line 114 at r1](https://reviewable.io:443/reviews/m-lab/etl/432#-L4mBkjm37DTS6_NPv4n:-L4mHYAppf9Fw5PGmILZ:b-zdriqb) ([raw file](https://github.com/m-lab/etl/blob/8713afb4a6f05ba27e6178386f76ed6dc0df17e3/.travis.yml#L114)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

What do you think about injecting things that differ between sandbox/staging/prod, instead of having separate yaml files?
I started this with gardener, and would like to work out whether to standardize on it.

</blockquote></details>

SGTM. I've removed the staging file and added env parameters to app-disco.yaml 

---

*[cmd/etl_worker/app-disco.yaml, line 42 at r1](https://reviewable.io:443/reviews/m-lab/etl/432#-L4mCJ7ONWANfgOJVLCx:-L4mErXfV3N4QzbX83PJ:b5gf1sg) ([raw file](https://github.com/m-lab/etl/blob/8713afb4a6f05ba27e6178386f76ed6dc0df17e3/cmd/etl_worker/app-disco.yaml#L42)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

base_tables?  Or incoming?
</blockquote></details>

Discussed in person. We'll use 'base_tables' for now until we move to 'incoming'.

---

*[cmd/etl_worker/app-disco-staging.yaml, line 42 at r1](https://reviewable.io:443/reviews/m-lab/etl/432#-L4mBy9TGMgkFD-1-RmG:-L4mF-N1WYbaDxpkrSPt:b84k10a) ([raw file](https://github.com/m-lab/etl/blob/8713afb4a6f05ba27e6178386f76ed6dc0df17e3/cmd/etl_worker/app-disco-staging.yaml#L42)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

I think we should actually use the same dataset in every project.  Different for batch vs daily, but otherwise the same across projects.  so - base_tables - Make sense?
</blockquote></details>

I agree. I've created `base_tables` dataset in mlab-sandbox and mlab-staging projects.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/432)*
<!-- Sent from Reviewable.io -->
